sdk/eventhub/Azure.Messaging.EventHubs/src/Producer/StreamingProducer.cs

```diff
@@ -0,0 +1,313 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+
+namespace Azure.Messaging.EventHubs.Producer
+{
+    /// <summary>
+    /// This class allows the application to create and use a Streaming
+    /// producer instance.
+    /// </summary>
+    public class StreamingProducer : IAsyncDisposable
+    {
+        private EventHubProducerClient Producer;
+        private ConcurrentDictionary<string, ConcurrentQueue<EventData>> PartitionQueues;
+        private ConcurrentDictionary<string, EventDataBatch> PendingBatches;
+        private ConcurrentQueue<EventData> GeneralQueue;
+        private EventDataBatch GeneralPendingBatch;
+
+        // <summary>
+        // A handler for event batches that have been sent successfully.
+        // </summary>
+        //public event Action<PublishSucceededEventArgs> PublishSucceeded;
+
+        // <summary>
+        // A handler for event batches that have failed to successfully send after applying the
+        // retry policy.
+        // </summary>
+        //public event Action<PublishFailedEventArgs> PublishFailed;
+
+        /// <summary>
+        /// The fully qualified Event Hubs namespace that this producer is currently associated with,
+        /// which will likely be similar to <c>{yournamespace}.servicebus.windows.net</c>.
+        /// </summary>
+        public string FullyQualifiedNamespace => Producer.FullyQualifiedNamespace;
+
+        /// <summary>
+        /// The name of the Event Hub that this producer is connected to, specific to the Event Hubs
+        /// namespace that contains it.
+        /// </summary>
+        public string EventHubName => Producer.EventHubName;
+
+        /// <summary>
+        /// </summary>
+        public string Identifier { get; }
```

Comment:
```suggestion
        public string Identifier => Producer.Identifier;
```

Once my changes are done, the Identifier can be set on the producer client options and we can just point to it.   (this will fail today, though.  😄  )